public/app/features/alerting/unified/hooks/useProduceNewRuleGroup.tsx

```diff
@@ -20,39 +28,19 @@ type ProduceResult = RulerRuleGroupDTO | AlertGroupUpdated;
  * and to guard against user concurrency issues.
  *
  * @throws
- * @TODO the manual state tracking here is not great, but I don't have a better idea that works /shrug
  */
 function useProduceNewRuleGroup() {
-  const [fetchRuleGroup] = alertRuleApi.endpoints.getRuleGroupForNamespace.useLazyQuery();
-  const [updateRuleGroup] = alertRuleApi.endpoints.updateRuleGroupForNamespace.useMutation();
-  const [deleteRuleGroup] = alertRuleApi.endpoints.deleteRuleGroupFromNamespace.useMutation();
-
-  const [isLoading, setLoading] = useState<boolean>(false);
-  const [isUninitialized, setUninitialized] = useState<boolean>(true);
-  const [result, setResult] = useState<ProduceResult | undefined>();
-  const [error, setError] = useState<unknown | undefined>();
-
-  const isError = Boolean(error);
-  const isSuccess = !isUninitialized && !isLoading && !isError;
-
-  const requestState = {
-    isUninitialized,
-    isLoading,
-    isSuccess,
-    isError,
-    result,
-    error,
-  };
+  const [fetchRuleGroup, requestState] = alertRuleApi.endpoints.getRuleGroupForNamespace.useLazyQuery();
 
   /**
    * This function will fetch the latest we have on the rule group, apply a diff to it via a reducer and sends
    * the new rule group update to the correct endpoint.
```

Comment:
we should update this comment as now we are not updating the group yet?

Reply:
Oops, I had upgraded the diagram but not the actual description 😅 good catch!